### PR TITLE
.NET 6 update for SignalR auth and authz /1

### LINKED ISSUE #24873: .NET 6: Authentication and authorization in ASP.NET Core SignalR
[Enter feedback here]

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d0da2144-3730-f8ef-1d16-a89f11fd6198
* Version Independent ID: 6d8f5bc8-8f5c-ba13-b7f2-0366eac46f81
* Content: [Authentication and authorization in ASP.NET Core SignalR](https://docs.microsoft.com/en-us/aspnet/core/signalr/authn-and-authz?view=aspnetcore-6.0)
* Content Source: [aspnetcore/signalr/authn-and-authz.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/signalr/authn-and-authz.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-signalr**
* GitHub Login: @bradygaster
* Microsoft Alias: **bradyg**

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #24873

[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/signalr/authn-and-authz?view=aspnetcore-6.0&branch=pr-en-us-24875)

### CODE REVIEW

File: aspnetcore/signalr/authn-and-authz/6.0sample/SignalRAuthenticationSample/DomainRestrictedRequirement.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+public class DomainRestrictedRequirement :
+    AuthorizationHandler<DomainRestrictedRequirement, HubInvocationContext>,
+    IAuthorizationRequirement
+{
+    protected override Task HandleRequirementAsync(AuthorizationHandlerContext context,
+        DomainRestrictedRequirement requirement,
+        HubInvocationContext resource)
+    {
+        var name = context.User.Identity!.Name!;
```
Comment (Rick-Anderson) [inline aspnetcore/signalr/authn-and-authz/6.0sample/SignalRAuthenticationSample/DomainRestrictedRequirement.cs:14]: @BrennanConroy  the ! operator is ugly here. I could use `#nullable disable` but that seems worse.

Comment (BrennanConroy) [reply aspnetcore/signalr/authn-and-authz/6.0sample/SignalRAuthenticationSample/DomainRestrictedRequirement.cs:14]:
```csharp
var name = context.User.Identity?.Name;
if (!string.IsNullOrEmpty(name) && ...)
{
    // ...
}
```

Comment (Rick-Anderson) [reply aspnetcore/signalr/authn-and-authz/6.0sample/SignalRAuthenticationSample/DomainRestrictedRequirement.cs]:
```
if (context.User.Identity != null && 
     !string.IsNullOrEmpty(context.User.Identity.Name))
        {
```

That looks worse to me but it's what I've pushed. An extra level of `{}`

Comment (BrennanConroy) [reply aspnetcore/signalr/authn-and-authz/6.0sample/SignalRAuthenticationSample/DomainRestrictedRequirement.cs]: I would have combined what I wrote with the if statement that was already there.
```csharp
var name = context.User.Identity?.Name;
if (!string.IsNullOrEmpty(name) && IsUserAllowedToDoThis(resource.HubMethodName, name) &&
    name.EndsWith("@microsoft.com"))
```

Comment (Rick-Anderson) [reply aspnetcore/signalr/authn-and-authz/6.0sample/SignalRAuthenticationSample/DomainRestrictedRequirement.cs]:
```
if (context.User.Identity != null &&
  !string.IsNullOrEmpty(context.User.Identity.Name) && 
  IsUserAllowedToDoThis(resource.HubMethodName,
                       context.User.Identity.Name) &&
  context.User.Identity.Name.EndsWith("@microsoft.com"))
{
        context.Succeed(requirement);
    
}
```
Too much in my opinion.

**Review by bradygaster — APPROVED** [review]
LGTM - no issues aside from those @BrennanConroy already called out. Wish there was a way it could be smaller - seems like a LOT of files and stuff for folks to wander through to learn but auth isn't easy. :)

**Comment by Rick-Anderson** [comment]
> Wish there was a way it could be smaller - seems like a LOT of files and stuff for folks to wander through to learn but auth isn't easy. :)

It's mostly template generated code which shouldn't need review.

**Pull request merged**